pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -115,15 +117,24 @@ public class RealtimeTableDataManager extends BaseTableDataManager {
 
   private TableDedupMetadataManager _tableDedupMetadataManager;
   private TableUpsertMetadataManager _tableUpsertMetadataManager;
+  // Object to track ingestion delay for all partitions
+  private IngestionDelayTracker _ingestionDelayTracker;
+  private Supplier<Boolean> _isReadyToServeQueries;
 
   public RealtimeTableDataManager(Semaphore segmentBuildSemaphore) {
+    this(segmentBuildSemaphore, () -> true);
+  }
+
+  public RealtimeTableDataManager(Semaphore segmentBuildSemaphore, Supplier<Boolean> isReadyToServeQueries) {
     _segmentBuildSemaphore = segmentBuildSemaphore;
+    _isReadyToServeQueries = isReadyToServeQueries;
```

Comment:
I suggest that the `isReadyToServeQueries` either be handled in the table data manager (by blocking all calls to ingestion delay tracker), or in the ingestion delay tracker itself.  Let us not involve the segment data manager with another state that it really does not need to know about

Reply:
This has been addressed in commit 2e4daa35609356522a483f2a9503dde263a2e5ff, please close this at your earliest if this satisfy your request